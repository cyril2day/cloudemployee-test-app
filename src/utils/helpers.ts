type ParamObject = {[name: string]: any }


export function normalize(parent: ParamObject) {
   if (parent && parent.children) {
      parent.children.forEach((child: any) => {
         if (!child.parentId)
            child.parentId = parent.id

         normalize(child)
      })
   }
}

export function flatten(items: ParamObject[]) {
   const temp: unknown[] = []

   items.forEach(item => {
      normalize(item)
      const { children, ...props } = item 

      temp.push(props)

      if (children && Array.isArray(children)) {
         temp.push(flatten(children))
      }
   })

   return temp.flat()
}
