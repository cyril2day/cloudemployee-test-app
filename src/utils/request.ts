import axios from 'axios'


const api = axios.create({ baseURL: 'https://cloudemployee-backend-test-app.vercel.app/api' })

api.interceptors.request.use(
   config => {
      return config
   }
)

api.interceptors.response.use(
   response => {
      if (response.data)
         return response
      else
         return Promise.reject(response)
   },
   error => Promise.reject(error.response)
)

export { api }
