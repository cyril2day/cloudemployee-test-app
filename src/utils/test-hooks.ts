import { MountingOptions } from "@vue/test-utils"
import { setActivePinia, createPinia } from 'pinia'

export type TestMountingOptions = MountingOptions<unknown>

export const StoreBeforeEach = () => {
   beforeEach(() => {
      setActivePinia(createPinia())
   })
}
