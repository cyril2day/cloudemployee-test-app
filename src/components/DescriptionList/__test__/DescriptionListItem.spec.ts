import { mount, VueWrapper } from "@vue/test-utils"
import { mergeWith } from "lodash"
import { TestMountingOptions } from "../../../utils/test-hooks"
import DescriptionListItemVue from "../DescriptionListItem.vue"


const createWrapper = (overrides?: TestMountingOptions) => {
   const defaultMountingOptions: TestMountingOptions = {
      global: {
         //
      }
   }

   return mount(DescriptionListItemVue, mergeWith(defaultMountingOptions, overrides))
}

describe('Description List Item Component Unit Test', () => {
   let wrapper: VueWrapper<any>

   beforeEach(() => {
      wrapper = createWrapper()
   })

   afterEach(() => {
      if (wrapper)
         wrapper.unmount()
   })

   test('component accepts five(5) slots)', () => {
      expect.assertions(1)
      const defaultNumOfProps = 5


      expect(Object.keys(wrapper.props()).length).toBe(defaultNumOfProps)
   })

   test('width props default value is 80', () => {
      expect.assertions(2)

      expect(wrapper.props('width')).toBe(80)
      expect(typeof wrapper.props('width')).toBe('number')
   })

   test('trim method works correctly', () => {
      expect.assertions(2)
      const sampleString = 'this is a long string',
         stringTrimmed = 'this ...',
         wrapper = createWrapper({
         props: {
            label: 'Sample Long String',
            value: sampleString,
            trim: 5 
         }
      })

      expect(wrapper.find('dd').element.textContent).toBe(stringTrimmed)
      expect(wrapper.find('dd').element.textContent?.length).toBe(stringTrimmed.length)
   })

   test('renders label and value props', () => {
      expect.assertions(2)
      const expectedLabel = 'Sample Label'
      const expectedParagraph = 'This is a sample paragraph'
      const wrapper = createWrapper({
         props: {
            label: expectedLabel,
            value: expectedParagraph
         }
      })
      
      expect(wrapper.find('dt').element.textContent).toBe(expectedLabel)
      expect(wrapper.find('dd').element.textContent).toBe(expectedParagraph)
   })
})


describe('Description List Item Component Snapshot Test', () => {
   test('Elements and styles renders correctly', () => {
      expect.assertions(1)
      const wrapper = createWrapper()

      expect(wrapper.element).toMatchSnapshot()
   })
})
