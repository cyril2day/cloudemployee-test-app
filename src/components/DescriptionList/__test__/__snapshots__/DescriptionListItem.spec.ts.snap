// Vitest Snapshot v1

exports[`Description List Item Component Snapshot Test > Elements and styles renders correctly 1`] = `
<div
  class="list-item__wrapper"
  data-v-f263b4d4=""
>
  <dt
    class="list-item__label"
    data-v-f263b4d4=""
  />
  <dd
    class="list-item__value"
    data-v-f263b4d4=""
  />
</div>
`;
